src/observers.ts

```diff
@@ -4,15 +4,26 @@ import { EventEmitter2 } from 'eventemitter2';
 import { isInputLikeElement, on, ResetHandler, toModifiers } from './util/fn';
 import { getSerializedDataTransferItemList } from './util/entry-reader';
 
-export interface IObserver {
+export interface IObserver<TOutput> {
   name: string;
   emitter: EventEmitter2;
   start(): void;
   stop(): void;
   suspend(): void;
+  on(listenerFn: ObserverListener<TOutput>): ObserverListener<TOutput>;
+  off(listenerFn: ObserverListener<TOutput>): void;
 }
 
-export abstract class AbstractObserver implements IObserver {
+export type ObserverListener<TOutput> = (output: TOutput) => void;
+
+export type EventProcessor<TEvent, TOutput> = (
+  event: TEvent,
+  ...args: any[]
```

Comment:
should be Serializable ?

Reply:
I think it depends on usage, it is not a mandatory requirement.